internal/exec/util/file.go

```diff
@@ -37,20 +35,24 @@ const (
 )
 
 type File struct {
-	io.ReadCloser
-	hash.Hash
+	Hash        hash.Hash
 	Path        string
 	Mode        os.FileMode
 	Uid         int
 	Gid         int
 	expectedSum string
+	Url         url.URL
+	Client      *resource.HttpClient
```

Comment:
The function names and comments no longer really lined up with what the logic was doing. I've renamed the functions and rewritten the comments, and also cleaned up the struct a little, so hopefully it's a little clearer now.

The `RenderFile` function is preparing a fetching operation to be performed, so it's now named `PrepareFetch` which returns a `FetchOp`. In this context, does it make sense why the URL, HttpClient, and compression are all necessary?